### ISSUE TITLE
Use psutil for reporting other running processes during testing

### ISSUE THREAD

**Issue by ronawho** [chapel-lang/chapel#5452]
For performance testing or when we have a timeout our testing system reports processes with the highest cpu utilization. This is intended to help us track down if regressions/timeouts were caused by a stray process/job.

For performance testing we do something like:

```bash
ps ax -o user,pid,pcpu,command | head -n 1 # print header
ps ax -o user,pid,pcpu,command | tail -n +2 | sort -r -k 3 | head -n 5 # print top 5 processes
```

and for timeouts (in timedexec) we do:

```bash
ps aux | sort -r -k 3 | head
```

Unfortunately this often results in errors about broken pipes for sort

```bash
sort: write failed: standard output: Broken pipe
sort: write error
```

which makes the output hard to parse:

```bash
USER        PID %CPU COMMAND
sort: write failed: standard output: Broken pipe
sort: write error
chapelu   33278  0.7 python /var/tmp/jenkins/chapel-ci/workspace/chapcs-perf-test-perf.chapcs/util/test/sub_test /var/tmp/jenkins/chapel-ci/workspace/chapcs-perf-test-perf.chapcs/bin/linux64/chpl
chapelu  137514  0.5 java -jar slave.jar
chapelu   36057  0.0 tail -n +2
chapelu  137168  0.0 sshd: chapelu@notty
root     137166  0.0 sshd: chapelu [priv]
```

I think the correct solution is to:
 - unify how we grab the top processes 
 - use [psutil] (https://pypi.python.org/pypi/psutil/) to grab results instead of shelling out
 - maybe only print out processes that are above some threshold of utilization (say 10%)

Possibly related: http://stackoverflow.com/questions/20573282/hudson-yes-standard-output-broken-pipe

**Comment by ronawho** [comment]
Copying a conversation from @bradcray about this:

>It fails under my testReleasesPerformance run for some reason as well. I
>always run it with 'nohup ...blah... 2>&1 &' -- I wonder if that could
>repro it on a simple run.
>
>Yeah, that seems to do it (on linux anyway... on mac it worked OK).
>
>-Brad
>
>
>>I don't think we had a fix for it, but I agree that it's super annoying.
>>
>>We do something like shell out and call `ps ax -o user,pid,pcpu,command
>>| sort -r -k 3 | head -n 5` in sub_test and for some reason that sort
>>fails. It seems to only fail under jenkins though, and not when I run
>>start_test manually. Maybe we can move the sorting into python instead
>>of shelling out
>>
>>Elliot
>>
>>>Hi Elliot --
>>>
>>>Remind me... what's the cause of this again?  (not that it's a problem
>>>I think, though it always makes me panic when I see it):
>>>
>>>[Reporting processes with top 5 highest cpu usages]
>>>USER        PID %CPU COMMAND
>>>sort: write failed: standard output: Broken pipe
>>>sort: write error
>>>chapelu  133036  0.5 java -jar slave.jar
>>>chapelu  115935  0.1 python /ptmp/relOverRel/chapel/util/start_test
>>>-performance --compopts --no-cc-warnings -num-trials 3
>>>chapelu  118855  0.0 tail -n +2
>>>chapelu   59892  0.0 sshd: chapelu@pts/0
>>>chapelu  132747  0.0 sshd: chapelu@notty
>>>
>>>
>>>and did we have a fix for it that never got implemented?
>>>
>>>Thanks,
>>>-Brad

**Comment by ronawho** [comment]
@bradcray `nohup start_test --performance test/studies/jacobi/ 2>&1 &` doesn't result in the broken pipe error for me. Do you remember what test you ran?

**Comment by bradcray** [comment]
I get it when I'm gathering release over release performance via ~/chapel/util/pastPerformance/testReleasesPerformance.  I just tried a bunch of things in my own workspace and file system to reproduce it similar to what I do in the release-over-release gathering and failed to reproduce it.  So that suggests it might be something specific to the chapelu account or the system/filesystem on which I ran it, or ...?
